### PR TITLE
feat: enable spring security

### PR DESCRIPTION
**Pull request by NurmukhametovAlexey**

### CODE REVIEW

File: src/main/java/ru/timur/learning/configuration/ApplicationInitializer.java [hunk 1]
```diff
@@ -1,27 +1,21 @@
...
-import javax.servlet.ServletContext;
-import javax.servlet.ServletRegistration;
-import org.springframework.web.WebApplicationInitializer;
-import org.springframework.web.context.ContextLoaderListener;
-import org.springframework.web.context.support.AnnotationConfigWebApplicationContext;
-import org.springframework.web.context.support.GenericWebApplicationContext;
-import org.springframework.web.servlet.DispatcherServlet;
+import org.springframework.web.servlet.support.AbstractAnnotationConfigDispatcherServletInitializer;
 
-public class ApplicationInitializer implements WebApplicationInitializer {
```
Comment (NurmukhametovAlexey) [inline src/main/java/ru/timur/learning/configuration/ApplicationInitializer.java:11]: In this file I didn't change anything - just used out-of-the-box AbstractAnnotationConfigDispatcherServletInitializer which does exactly the same work you did manually before

File: src/main/java/ru/timur/learning/configuration/SecurityWebApplicationInitializer.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package ru.timur.learning.configuration;
+
+import org.springframework.security.web.context.AbstractSecurityWebApplicationInitializer;
+
+public class SecurityWebApplicationInitializer extends AbstractSecurityWebApplicationInitializer {
```
Comment (NurmukhametovAlexey) [inline src/main/java/ru/timur/learning/configuration/SecurityWebApplicationInitializer.java:5]: That's what was lacking for security to work - I lied the week before, everything should work now, even non-default stuff. 🌵

**Comment by tyambulatov** [comment]
Excellent. It was great that you were able to solve the problem. 
I haven't even considered a different `ApplicationInitializer` implementation.

**Review by tyambulatov — APPROVED** [review]
All good

**Pull request merged**